pinot-core/src/main/java/com/linkedin/pinot/core/startree/hll/ArrayConverter.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.startree.hll;
+
+/**
+ * Convert between byte array and char array, one byte is mapped to one char and vice versa.
+ * This is due to UTF-8 encoding for String type serialization used all over the system.
+ */
+public class ArrayConverter {
+
+    public static char[] byteArrayToChars(byte[] byteArray) {
+        char[] charArrayBuffer = new char[byteArray.length];
+        for (int i = 0; i < byteArray.length; i++) {
+            charArrayBuffer[i] = byteToChar(byteArray[i]);
+        }
+        return charArrayBuffer;
+    }
+
+    public static byte[] charsToByteArray(char[] chars) {
+        byte[] ret = new byte[chars.length];
```

Comment:
It's better to wrap this as CharBuffer and then get a byte array out of it. UTF-8 can be > 1 byte.